src/chapter4/multithreading.md

```diff
@@ -1 +1,54 @@
-# Multithreading
+# Multithreading on HPC
+
+## Thread vs Process
+
+![](src/chapter4/_attachments/Pasted%20image%2020230325112805.png)
+
+When computer runs a program, your source code is loaded into RAM and process is started.
+A **process** is a collection of code, memory, data and other resources.
+A process runs in a unique address space. So Two processes can not see each other’s memory. 
+
+A **thread** is a sequence of code that is executed inside the scope of the **process**. You can (usually) have multiple **threads** executing concurrently within the same process. 
+**Threads**  can view the memory (i.e. variables) of other threads within the same process
+
+A **multiprocessing** system has more than two processors, whereas **multithreading** is a program execution technique that allows a single process to have multiple code segments. 
+
+## Architecture of a HPC Cluster (Massive)
+
+![](src/chapter4/_attachments/Pasted%20image%2020230326141219.png)
+
+The key in HPC is to write a parallel computing code that utilise multiple nodes at the same time. essentially, more computers faster your application
+
+## Using Massive
+
+### Find Available Partition
+
+command: `show_cluster`
+
+![](src/chapter4/_attachments/Pasted%20image%2020230326141406.png)
```

Comment:
Maybe save images to `imgs/` directory. You can also use relative paths rather than the absolute path. I also then the image needs a label/name.